main/webapp/modules/core/scripts/project/history-panel.js

```diff
@@ -288,9 +288,35 @@ HistoryPanel.prototype._showApplyOperationsDialog = function() {
   
   elmts.dialogHeader.html($.i18n('core-project/apply-operation'));
   elmts.or_proj_pasteJson.html($.i18n('core-project/paste-json'));
+
+  elmts.operationJsonButton.on('click', async function() {
+    const fileInput = elmts.operationJsonButton[0];
+    fileInput.accept = '.json';
+    fileInput.onchange = async function() {
+      const file = fileInput.files[0];
+      const reader = new FileReader();
+      reader.onload = function(e) {
+        try {
+          const fileContent = JSON.parse(e.target.result);
+          const textAreaElement = elmts.textarea[0];
+          if (textAreaElement) {
+            textAreaElement.textContent = JSON.stringify(fileContent, null, 2)
+          }
+        } catch (error) {
+          {
+            window.alert($.i18n('core-index-import/warning-data-file'));
+          }
```

Comment:
The `{` and `}` around `window.alert(...)` should not be necessary: no need to duplicate those of the `catch` block.

Reply:
Got it I will fix it, used something that existed from the docs, but I will create a new one for this.

Reply:
Looks good to me, if you could just remove the `+"."` at the end it would be great because there might be languages where that does not make sense.